src/syscheckd/db/include/fimDB.hpp

```diff
@@ -306,6 +298,10 @@ class FIMDB
             m_stopping = true;
         };
 
+        inline void loggingFunction(modules_log_level_t logType, const std::string& msg){
+            this->m_loggingFunction(logType, msg);
```

Comment:
This looks good, but we must sanitize in FIMDB::init that it was not initialized with a pointer to null (The function pointer that is called inside the std::function). If it is mandatory to have a logging function, please throw exception in the constructor with its sanitation.

Reply:
This function will be deleted from this PR as i'ts already included in this [PR](https://github.com/wazuh/wazuh/pull/11307)